lib/core/api.js

```diff
@@ -359,8 +359,13 @@ function addDataToDataset(data, datasetId, blockIndex, uploadType, callback) {
       callback('duplicate', jqxhr.responseJSON.index);
     } else {
       api.log('platform add data to dataset failed.');
-      api.log(status);
-      callback(status);
+      var err = new Error('Uploading data to platform failed.');
+      err.name = 'API Error';
+      err.status = status;
+      err.datasetId = datasetId;
+      err.sessionId = status.meta.trace.request;
```

Comment:
Done.

https://trello.com/c/aWY4xoU0/271-0-ux-dev-0-tst-uploader-add-platform-x-tidepool-trace-session-header-to-all-api-requests-for-improved-tracking-of-errors <https://trello.com/c/aWY4xoU0/271-0-ux-dev-0-tst-uploader-add-platform-x-tidepool-trace-session-header-to-all-api-requests-for-improved-tracking-of-errors>

> On Aug 2, 2016, at 3:47 AM, Gerrit Niezen <notifications@github.com> wrote:
> 
> In lib/core/api.js <https://github.com/tidepool-org/chrome-uploader/pull/312#discussion_r73133602>:
> 
> > @@ -359,8 +359,13 @@ function addDataToDataset(data, datasetId, blockIndex, uploadType, callback) {
> >        callback('duplicate', jqxhr.responseJSON.index);
> >      } else {
> >        api.log('platform add data to dataset failed.');
> > -      api.log(status);
> > -      callback(status);
> > +      var err = new Error('Uploading data to platform failed.');
> > +      err.name = 'API Error';
> > +      err.status = status;
> > +      err.datasetId = datasetId;
> > +      err.sessionId = status.meta.trace.request;
> @darinkrauss <https://github.com/darinkrauss> Will you make a card on the backlog to add the session ID so that we don't forget?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/tidepool-org/chrome-uploader/pull/312/files/c8f5682c3542a07e571164e92872c0b46bf1e6d7#r73133602>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ABKu8cw6SUPKMuTPuU6YFy58dO81zMBgks5qbyBPgaJpZM4JZmMb>.
>